libtsuba/src/RDG.cpp

```diff
@@ -81,45 +81,29 @@ WriteProperties(
 }
 
 katana::Result<void>
-CommitRDG(
+ComputeManifest(
     katana::RDGHandle handle, uint32_t policy_id, bool transposed,
     katana::RDG::RDGVersioningPolicy versioning_action,
-    const katana::RDGLineage& lineage,
-    std::unique_ptr<katana::WriteGroup> desc) {
+    const katana::RDGLineage& lineage, std::unique_ptr<katana::WriteGroup> desc,
+    katana::TxnContext* txn_ctx) {
   katana::CommBackend* comm = katana::Comm();
   katana::RDGManifest new_manifest =
       (versioning_action == katana::RDG::RetainVersion)
           ? handle.impl_->rdg_manifest().SameVersion(
                 comm->Num, policy_id, transposed, lineage)
           : handle.impl_->rdg_manifest().NextVersion(
                 comm->Num, policy_id, transposed, lineage);
+  txn_ctx->SetManifest(new_manifest);
 
   // wait for all the work we queued to finish
-  TSUBA_PTP(katana::internal::FaultSensitivity::High);
   KATANA_CHECKED_CONTEXT(desc->Finish(), "at least one async write failed");
 
-  TSUBA_PTP(katana::internal::FaultSensitivity::High);
-  comm->Barrier();
-
-  TSUBA_PTP(katana::internal::FaultSensitivity::High);
-  katana::Result<void> ret = katana::OneHostOnly([&]() -> katana::Result<void> {
-    TSUBA_PTP(katana::internal::FaultSensitivity::High);
-
-    std::string curr_s = new_manifest.ToJsonString();
-    auto manifest_file = katana::RDGManifest::FileName(
-        handle.impl_->rdg_manifest().dir(),
-        handle.impl_->rdg_manifest().viewtype(), new_manifest.version());
-    KATANA_CHECKED_CONTEXT(
-        katana::FileStore(
-            manifest_file.string(),
-            reinterpret_cast<const uint8_t*>(curr_s.data()), curr_s.size()),
-        "CommitRDG future failed {}", manifest_file);
-    return katana::ResultSuccess();
-  });
-  if (ret) {
-    handle.impl_->set_rdg_manifest(std::move(new_manifest));
-  }
-  return ret;
+  auto manifest_file = katana::RDGManifest::FileName(
```

Comment:
This request is unrelated to your PR, but can you mark this with a TODO(witchel) that the file names generated by `RDGManifest::FileName` are not correct because they do not include the partitioning policy and the number of partitioned hosts.  I recently discovered this problem trying to debug `tsuba_gc`.  Unfortunately, our manifest naming logic is kind of screwed up right now.